src/api_server/swagger/firecracker.yaml

```diff
@@ -699,6 +811,19 @@ definitions:
         description: The amount of milliseconds it takes for the bucket to refill.
         minimum: 0
 
+  Vm:
+    type: object
+    description:
+      Defines the microVM state. It is especially useful in the snapshotting context.
+    required:
+      - state
+    properties:
+      state:
+        type: string
+        enum:
+          - Paused
+          - Resumed
```

Comment:
This is the same design as `/actions` where we have an enum stating the action to do. `pause` and `resume` can go there to keep things consistent.

The reason https://github.com/firecracker-microvm/firecracker/pull/1537 avoids using `actions` is that the design there is not future-proof/flexible. It does not naturally support any _actions_ that have a body (extra data like identifiers or payload or such).
You can check the infamous `BlockDeviceRescan` action in past Firecracker versions code, which made the code a mess by introducing a `payload` next to the enum which had to be validated and interpreted differently depending on the enum variant (even for other actions which hadn't been aware of payloads).

Reply:
Or if we feel it's more RESTful, all `actions` can go under `/vm`